core/modules/replica/testConfiguration.cc

```diff
@@ -52,479 +56,380 @@ BOOST_AUTO_TEST_CASE(ConfigurationTest) {
+    vector<string> workers4;
+    BOOST_REQUIRE_NO_THROW(workers4 = config->workers(isEnabled));
+    sort(workers4.begin(), workers4.end());
+    BOOST_CHECK(workers4.size() == 1);
+    BOOST_CHECK(workers4 == vector<string>({"worker-C"}));
+
+    // Fetching values of general parameters.
+    BOOST_CHECK(config->get<size_t>("common", "request_buf_size_bytes") == 8192);
+    BOOST_CHECK(config->get<unsigned int>("common", "request_retry_interval_sec") == 1);
+
+    BOOST_CHECK(config->get<size_t>("controller", "num_threads") == 2);
+    BOOST_CHECK(config->get<uint16_t>("controller", "http_server_port") == 8080);
+    BOOST_CHECK(config->get<size_t>("controller", "http_server_threads") == 3);
+    BOOST_CHECK(config->get<unsigned int>("controller", "request_timeout_sec") == 100);
+    BOOST_CHECK(config->get<string>("controller", "empty_chunks_dir") == "/qserv/data/qserv");
+    BOOST_CHECK(config->get<unsigned int>("controller", "job_timeout_sec") == 200);
+    BOOST_CHECK(config->get<unsigned int>("controller", "job_heartbeat_sec") == 300);
+
+    BOOST_CHECK(config->get<unsigned int>("xrootd", "auto_notify") == 0);
+    BOOST_CHECK(config->get<string>("xrootd", "host") == "localhost");
+    BOOST_CHECK(config->get<uint16_t>("xrootd", "port") == 1104);
+    BOOST_CHECK(config->get<unsigned int>("xrootd", "request_timeout_sec") == 400);
+
+    BOOST_CHECK(config->get<string>("database", "host") == "localhost");
+    BOOST_CHECK(config->get<uint16_t>("database", "port") == 13306);
+    BOOST_CHECK(config->get<string>("database", "user") == "qsreplica");
+    BOOST_CHECK(config->get<string>("database", "password") == "changeme");
+    BOOST_CHECK(config->get<string>("database", "name") == "qservReplica");
+
+    BOOST_CHECK(config->get<string>("database", "qserv_master_host") == "localhost");
+    BOOST_CHECK(config->get<uint16_t>("database", "qserv_master_port") == 3306);
+    BOOST_CHECK(config->get<string>("database", "qserv_master_user") == "qsmaster");
+    BOOST_CHECK(config->qservMasterDatabasePassword().empty());
+    BOOST_CHECK(config->get<string>("database", "qserv_master_name") == "qservMeta");
+
+    BOOST_CHECK(config->get<size_t>("database", "services_pool_size") == 2);
+
+    // Selecting and probing database families.
+    vector<string> families;
+    BOOST_REQUIRE_NO_THROW(families = config->databaseFamilies());
+    sort(families.begin(), families.end());
+    BOOST_CHECK(families.size() == 2);
+    BOOST_CHECK(families == vector<string>({"production", "test"}));
+    for (auto&& name: families) {
+        BOOST_CHECK(config->isKnownDatabaseFamily(name));
+    }
+    DatabaseFamilyInfo production;
+    BOOST_REQUIRE_NO_THROW(production = config->databaseFamilyInfo("production"));
+    BOOST_CHECK(production.name == "production");
+    BOOST_CHECK(production.replicationLevel == 10);
+    BOOST_CHECK(production.numStripes == 11);
+    BOOST_CHECK(production.numSubStripes == 12);
+    BOOST_CHECK(production.overlap == 0.01667);
```

Comment:
Exact comparison for floating point numbers can break (especially for fractional numbers).

Reply:
Well, then I should probably use `epsilon` for this case. I use this technique in some other code in a similar context:
```
std::abs(familyInfo.overlap - overlap) <= std::numeric_limits<double>::epsilon())
```

Reply:
`epsilon` is relative, so it's going to work if values subtracted are close to 1, if they are large then you need to scale it. BOOST has some useful tools for floating point comparison, check for `tolerance` on this page: https://www.boost.org/doc/libs/1_63_0/libs/test/doc/html/boost_test/testing_tools/boost_test_universal_macro.html